### PR TITLE
Cosmetic cleanup SIM interface to field

### PR DESCRIPTION
**Pull request by civanch**
Modifications in Field: use local  mutable variable instead of tls, reduce number of operations and remove unnecessary checks. This method is called several times at a simulation step, so minor optimisation is useful.  

In FieldBuilder a LogInfo with the full list of simulation parameters is added.

Overall should bring a tiny speadup.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4Core/MagneticField/src/Field.cc [hunk 1]
```diff
@@ -1,74 +1,48 @@
...
+  for(int i=0; i<3; ++i) {
+    oldx[i] = 1.0e12;
+    oldb[i] = 0.0;
+  }
 }
 
 Field::~Field() {}
 
-void Field::GetFieldValue(const double xyz[3],double bfield[3]) const 
+void Field::GetFieldValue(const G4double xyz[4], double* bfield) const
```
Comment (davidlange6) [inline SimG4Core/MagneticField/src/Field.cc]: Hi @civanch - i don't consider the change from double[3] to double* to be an improvement in the interface (quite the opposite). Why do you propose this change?

Comment (civanch) [inline SimG4Core/MagneticField/src/Field.cc]: @davidlange6 , I simply implement exact interface of G4MagneticField. Few years ago G4 interface was changed. In this CMS class the old version is implemented. However, both are equivalent and this change is not important. The main modifications concern removal of unnecessary operations which were were performed several times at each simulation step: call to tls and check on nan.
More serious modification for magnetic field is in  a progress so , please, merge this PR, so I can propose the next one.

Comment (civanch) [inline SimG4Core/MagneticField/src/Field.cc]: you mean to change second parameter double bfield[3] ? I do not think we should switch to std::array until Geant4 developers will do that in the base interface.

**Comment by davidlange6** [comment]
ah - is this being called by a G4 function (or is it calling G4)?

**Comment by civanch** [comment]
It is called by G4.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
@davidlange6 , I make the use of this interface more explicit.

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**